app/models/website.rb

```diff
@@ -18,14 +18,31 @@ class Website < ActiveRecord::Base
 
   attr_accessor :generate
 
+  before_validation :check_origins
   before_create :generate_api_keys
   before_update :generate_api_keys, if: -> { generate }
+  before_update :check_required
   before_destroy :website_dependent
 
   def website_dependent
     website_members.each(&:delete)
   end
 
+  def check_origins
+    return true if origins == '*'
+    schemes = ["http://", "https://", "ftp://", "ftps://", "sftp://"]
+    origins.split("\n").each do |origin|
+      unless schemes.any? { |scheme| origin.include? scheme }
+        errors.add(:origins, 'Please enter a valid origin')
```

Comment:
We need to be more specific with the error. For example in this case we are checking if the urls include a scheme. So the error should be `Please add origin urls with http/https etc.`